File: src/main/java/com/jss/osiris/modules/osiris/reporting/service/IncidentCauseService.java
package com.jss.osiris.modules.osiris.reporting.service;

import java.util.List;

import com.jss.osiris.modules.osiris.reporting.model.IncidentCause;

public interface IncidentCauseService {
    public List<IncidentCause> getIncidentCauses();

    public IncidentCause getIncidentCause(Integer id);

    public IncidentCause addOrUpdateIncidentCause(IncidentCause incidentCause);
}

File: client/src/app/modules/reporting/services/incident.cause.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { IncidentCause } from '../../reporting/model/IncidentCause';

@Injectable({
  providedIn: 'root'
})
export class IncidentCauseService extends AppRestService<IncidentCause>{

  constructor(http: HttpClient) {
    super(http, "reporting");
  }

  getIncidentCauses() {
    return this.getList(new HttpParams(), "incident-causes");
  }
  
   addOrUpdateIncidentCause(incidentCause: IncidentCause) {
    return this.addOrUpdate(new HttpParams(), "incident-cause", incidentCause, "Enregistré", "Erreur lors de l'enregistrement");
  }

}
